### ISSUE TITLE
F821 false positive with `BaseExceptionGroup` + `sys.version_info >= (3, 11)`

### ISSUE THREAD

**Issue by Liam-DeVoe** [astral-sh/ruff#22432]
### Summary

```python
import sys

if sys.version_info >= (3, 11):
    _ = BaseExceptionGroup
```

`ruff check` reports:

```
F821 Undefined name `BaseExceptionGroup`. Consider specifying `requires-python = ">= 3.11"` or `tool.ruff.target-version = "py311"` in your `pyproject.toml` file.
 --> sandbox.py:4:9
  |
3 | if sys.version_info >= (3, 11):
4 |     _ = BaseExceptionGroup
  |         ^^^^^^^^^^^^^^^^^^
  |

Found 1 error.
```

`BaseExceptionGroup` was added in 3.11, and shouldn't be reported as an error.

I don't know if ruff guarantees this kind of static analysis for `sys.version_info` checks. IIRC I've seen ruff be smart about other version-guarded analysis, which is why I'm filing this as a bug report. If not, consider this a feature request :-) 

### Version

ruff 0.14.10

**Comment by MichaReiser** [comment]
Thanks, this is  https://github.com/astral-sh/ruff/issues/13296
